grommet/hpe-design-system
# Example component - Always display figcaption below example container

[Deploy Preview](https://deploy-preview-2224--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Modifies how the figcaption and exampleControls are built up. Previously, if trying to use both `caption` and example controls (figma link, grommet links, etc.), the caption would be inserted between the example and the controls, this ensures the caption is displayed below the example container.

#### Where should the reviewer start?

Example.js

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**

**Accessibility Checks**

**Code Quality Checks**

#### How should this be manually tested?

Modify an Example displaying controls, add a caption, then verify caption placement visually and within the DOM (<figcaption> should be last child of <figure>).

For example:
```
<Example
  code={[
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/dashboards/DashboardExample.js",  
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/dashboards/components/DashboardGrid.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/dashboards/components/DashboardCard.js", 
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/global-header/components/GlobalHeader.js',
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/dashboards/components/data.js", 
    ]}
  template
  showResponsiveControls={['laptop', 'mobile']}
  width="100%"
  caption="test"
>
  <DashboardExample />
</Example>
```

#### Any background context you want to provide?

#### What are the relevant issues?

#### Screenshots (if appropriate)

#### Should this PR be mentioned in Design System updates?

#### Is this change backwards compatible or is it a breaking change?
